diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -305,6 +309,26 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
                     getAllLeafsWithSpecificType(SAFE_ACCESS, it)
                 }
             }
+            // fixme: we can't distinguish fully qualified names from chain of calls for now
+            ?.filter {
```

Comment:
Maybe, this logic can be done as `dropWhile` on list of `.` and `?.`? This way you won't need to split expressions once again for every element.